kernel/indTyping.ml

```diff
@@ -200,28 +200,88 @@ let unbounded_from_below u cstrs =
 let template_polymorphic_univs ~ctor_levels uctx paramsctxt concl =
   let check_level l =
     Univ.LSet.mem l (Univ.ContextSet.levels uctx) &&
+    (let () = assert (not @@ Univ.Level.is_small l) in true) &&
     unbounded_from_below l (Univ.ContextSet.constraints uctx) &&
     not (Univ.LSet.mem l ctor_levels)
```

Comment:
The `let assert in true` looks surprising, maybe 
```suggestion
    Univ.LSet.mem l (Univ.ContextSet.levels uctx) &&
    assert (not @@ Univ.Level.is_small l);
    unbounded_from_below l (Univ.ContextSet.constraints uctx) &&
    not (Univ.LSet.mem l ctor_levels)
```
instead

Reply:
Does this even parse or type-check?